### PR TITLE
Convert CUDART spec to C++ and generate C++ files

### PR DESCRIPTION
**Pull request by yuhc**

### CODE REVIEW

File: cava/samples/cudart/cudart.cpp [hunk 1]
```diff
@@ -137,14 +139,14 @@ ava_utility int __helper_cubin_num(void **cubin_handle) {
 ava_utility void __helper_dump_fatbin(void *fatCubin,
                                     GHashTable **fatbin_funcs,
                                     int *num_funcs) {
-    struct fatbin_wrapper *wp = fatCubin;
+    struct fatbin_wrapper *wp = (struct fatbin_wrapper *)fatCubin;
```
Comment (photoszzt) [inline cava/samples/cudart/cudart.cpp:142]: static_cast is better.

File: cava/samples/cudart/cudart.cpp [hunk 2]
```diff
@@ -226,7 +228,7 @@ ava_utility void __helper_dump_fatbin(void *fatCubin,
 }
 
 ava_utility void __helper_print_fatcubin_info(void *fatCubin, void **ret) {
-    struct fatbin_wrapper *wp = fatCubin;
+    struct fatbin_wrapper *wp = (struct fatbin_wrapper *)fatCubin;
```
Comment (photoszzt) [inline cava/samples/cudart/cudart.cpp:231]: same as above

File: cava/samples/cudart/cudart.cpp [hunk 3]
```diff
@@ -240,7 +242,7 @@ ava_utility void __helper_print_fatcubin_info(void *fatCubin, void **ret) {
     int i, j;
     for (i = 0; i < 100; i++)
         if (fatBinaryEnd[i] == 0x7F && fatBinaryEnd[i+1] == 'E' && fatBinaryEnd[i+2] == 'L') {
-            printf("ELF header appears at 0x%d (%p): \n", i, (void *)wp->ptr + i);
+            printf("ELF header appears at 0x%d (%lx): \n", i, (uintptr_t)wp->ptr + i);
```
Comment (photoszzt) [inline cava/samples/cudart/cudart.cpp:245]: same

File: cava/samples/cudart/cudart.cpp [hunk 4]
```diff
@@ -288,7 +291,7 @@ __cudaRegisterFatBinary(void *fatCubin)
 
     if (ava_is_worker) {
         //__helper_print_fatcubin_info(fatCubin, ret);
-        __helper_init_module(fatCubin, ret);
+        __helper_init_module((struct fatbin_wrapper *)fatCubin, ret);
```
Comment (photoszzt) [inline cava/samples/cudart/cudart.cpp:294]: static_cast

File: cava/samples/cudart/cudart.cpp (hunk 1 above)
Comment (yuhc) [reply cava/samples/cudart/cudart.cpp:143]: `static_cast<type>(value)` syntax is not parse-able by CAvA yet. Leave it as TODO.

**Pull request merged**